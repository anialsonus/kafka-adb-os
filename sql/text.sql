-- Test 'text' deserialization
-- These tests do not require a Kafka instance. They can be run as common tests.

-- start_matchignore
-- m/^NOTICE:  Kafka-ADB: Offset for partition [0-9]* is not known, and is set to default value [0-9]*.*/
-- end_matchignore

-- start_ignore
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;

DROP EXTENSION IF EXISTS kadb_fdw CASCADE;
CREATE EXTENSION kadb_fdw;

CREATE SERVER test_kadb_fdw_distribution_server
FOREIGN DATA WRAPPER kadb_fdw
OPTIONS (
    k_brokers '0.0.0.0:9092'
);
-- end_ignore


-- Test: CREATE TABLE with format 'text'

CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: ERROR for a SELECT from a table with more than one column

-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(a TEXT, b JSON, c INT, d TEXT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT a, b, c, d FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: ERROR for a SELECT from a table with one normal and one dropped column

-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(a TEXT, b JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);

ALTER TABLE test_kadb_fdw_text DROP COLUMN a;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT b FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: ERROR for a SELECT from a table with a single dropped column

-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);

ALTER TABLE test_kadb_fdw_text DROP COLUMN j;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT * FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: Normal SELECT, JSON data

-- start_ignore

CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT j FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: Normal SELECT, INT data

-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(i INT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '42'
);

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT i FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore


-- Test: Normal SELECT, empty data

-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(i INT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',

    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',

    k_tuples_per_partition_on_inject '1',
    text_data_on_inject ''
);

SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
-- end_ignore

SELECT i FROM test_kadb_fdw_text;

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;

SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
-- end_ignore
