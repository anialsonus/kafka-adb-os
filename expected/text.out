-- Test 'text' deserialization
-- These tests do not require a Kafka instance. They can be run as common tests.
-- start_matchignore
-- m/^NOTICE:  Kafka-ADB: Offset for partition [0-9]* is not known, and is set to default value [0-9]*.*/
-- end_matchignore
-- start_ignore
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
NOTICE:  extension "gp_inject_fault" already exists, skipping
DROP EXTENSION IF EXISTS kadb_fdw CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to server test_kadb_fdw_distribution_server
drop cascades to foreign table test_kadb_fdw_t
CREATE EXTENSION kadb_fdw;
CREATE SERVER test_kadb_fdw_distribution_server
FOREIGN DATA WRAPPER kadb_fdw
OPTIONS (
    k_brokers '0.0.0.0:9092'
);
-- end_ignore
-- Test: CREATE TABLE with format 'text'
CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);
-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: ERROR for a SELECT from a table with more than one column
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(a TEXT, b JSON, c INT, d TEXT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT a, b, c, d FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
ERROR:  Kafka-ADB: 'text' format can only be applied to a table with a single attribute (column)  (seg2 slice1 127.0.1.1:6004 pid=51954)
-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: ERROR for a SELECT from a table with one normal and one dropped column
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(a TEXT, b JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);
ALTER TABLE test_kadb_fdw_text DROP COLUMN a;
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT b FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
ERROR:  Kafka-ADB: 'text' format can only be applied to a table with a single attribute (column)  (seg1 slice1 127.0.1.1:6003 pid=51953)
-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: ERROR for a SELECT from a table with a single dropped column
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);
ALTER TABLE test_kadb_fdw_text DROP COLUMN j;
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT * FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
ERROR:  Kafka-ADB: 'text' format can only be applied to a table with a single attribute (column)  (seg0 slice1 127.0.1.1:6002 pid=51952)
-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: Normal SELECT, JSON data
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(j JSON)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '{"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}'
);
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT j FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
                                   j                                    
------------------------------------------------------------------------
 {"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}
 {"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}
 {"a": "Letter A", "b": {"b1": 1, "b2": "Field B2"}, "c": 3, "d": null}
(3 rows)

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: Normal SELECT, INT data
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(i INT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject '42'
);
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT i FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
 i  
----
 42
 42
 42
(3 rows)

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
-- Test: Normal SELECT, empty data
-- start_ignore
CREATE FOREIGN TABLE test_kadb_fdw_text(i INT)
SERVER test_kadb_fdw_distribution_server
OPTIONS (
    format 'text',
    k_topic 'test_topic',
    k_consumer_group 'test_consumer_group',
    k_seg_batch '3',
    k_timeout_ms '1000',
    k_tuples_per_partition_on_inject '1',
    text_data_on_inject ''
);
SELECT gp_inject_fault_infinite('kadb_fdw_inject_tuples', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_partition_1_per_segment', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

SELECT gp_inject_fault_infinite('kadb_fdw_inject_text', 'skip', dbid)
FROM gp_segment_configuration;
 gp_inject_fault_infinite 
--------------------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
SELECT i FROM test_kadb_fdw_text;
NOTICE:  Kafka-ADB: Offset for partition 0 is not known, and is set to default value 0  (seg0 slice1 127.0.1.1:6002 pid=51952)
NOTICE:  Kafka-ADB: Offset for partition 2 is not known, and is set to default value 0  (seg2 slice1 127.0.1.1:6004 pid=51954)
NOTICE:  Kafka-ADB: Offset for partition 1 is not known, and is set to default value 0  (seg1 slice1 127.0.1.1:6003 pid=51953)
 i 
---
  
  
  
(3 rows)

-- start_ignore
DROP FOREIGN TABLE test_kadb_fdw_text;
SELECT gp_inject_fault('all', 'reset', dbid)
FROM gp_segment_configuration;
 gp_inject_fault 
-----------------
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
 Success:
(8 rows)

-- end_ignore
